File: josephus.py
def josephus(n, k):
    res = 0
    for i in range(1, n+1):
        res = (res + k) % i
    return res + 1

if __name__ == "__main__":
    n, k = 7, 3
    print(f"Josephus({n}, {k}) =", josephus(n, k))

File: josephus.c
#include <stdio.h>

int josephus(int n, int k) {
    int res = 0; // 0-based
    for (int i = 1; i <= n; ++i)
        res = (res + k) % i;
    return res + 1; // convert to 1-based
}

int main() {
    int n = 7, k = 3;
    printf("Josephus(%d, %d) = %d\n", n, k, josephus(n, k));
    return 0;
}
